client/src/modules/list/components/Items/model/reducer.js

```diff
@@ -162,16 +189,38 @@ const items = (state = {}, action) => {
       return state;
     }
     case ItemActionTypes.MARK_AS_DONE_SUCCESS: {
-      const { itemId, editedBy } = action.payload;
-      const item = state[itemId];
+      const { itemId, editedBy, listState } = action.payload;
+      const item = { ...state[itemId], editedBy, done: true };
+      const items = { ...state, [itemId]: item };
 
-      return { ...state, [itemId]: { ...item, editedBy, done: true } };
+      return {
+        ...items,
+        [itemId]: {
+          ...items[itemId],
+          animate: shouldAnimate(items, item, listState)
+        }
+      };
```

Comment:
it can be simpler

```
return {
  ...state,
  [itemId]: {
    ...state[itemId],
    editedBy,
    done: true,
    animate: shouldAnimate(items, item, listState)
  }
};
```